resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/PerformanceCriticalCodeStageUtil.cs

```diff
@@ -166,11 +165,17 @@ public static bool IsPerformanceCriticalRootMethod(ITreeNode node)
             if (typeElement == null)
                 return false;
 
-            if (!typeElement.DerivesFromMonoBehaviour())
-                return false;
+            if (!typeElement.DerivesFromMonoBehaviour() && declaration.DeclaredElement is IClrDeclaredElement monoBehaviorCLRDeclaredElement)
+                return ourKnownHotMonoBehaviourMethods.Contains(monoBehaviorCLRDeclaredElement.ShortName);
```

Comment:
Are these `if` checks correct? This is looking to see if the type element does NOT derive from `MonoBehaviour` but then checks against the `MonoBehaviour` lookups. Same for all of the checks here.